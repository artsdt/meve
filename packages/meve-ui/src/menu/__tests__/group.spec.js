import Menu from '..'
import MenuItemGroup from '../../menu-item-group'
import MenuItem from '../../menu-item'
import { mount } from '@vue/test-utils'

test('test menu item group props', async () => {
  const component = {
    components: {
      [Menu.name]: Menu,
      [MenuItemGroup.name]: MenuItemGroup,
      [MenuItem.name]: MenuItem,
    },
    template: `
      <m-menu>
        <m-menu-item-group name="1" label="level 1" icon="error">
          <m-menu-item name="1-1" label="level 1-1" icon="error" />
        </m-menu-item-group>
      </m-menu>
    `,
  }

  const wrapper = mount(component)
  expect(wrapper.html()).toMatchSnapshot()
  wrapper.destroy()
})

test('test menu item group disabled', async () => {
  const handleGroupChange = jest.fn()

  const component = {
    components: {
      [Menu.name]: Menu,
      [MenuItemGroup.name]: MenuItemGroup,
      [MenuItem.name]: MenuItem,
    },
    methods: {
      handleGroupChange,
    },
    data: () => ({
      disabled: true,
    }),
    template: `
      <m-menu @update:expanded-names="handleGroupChange">
        <m-menu-item-group name="1" label="level 1" icon="error" :disabled="disabled">
          <m-menu-item name="1-1" label="level 1-1" icon="error" />
        </m-menu-item-group>
      </m-menu>
    `,
  }

  const wrapper = mount(component)
  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.find('.m-menu-item-group__trigger').trigger('click')
  expect(handleGroupChange).toHaveBeenCalledTimes(0)

  await wrapper.setData({ disabled: false })

  await wrapper.find('.m-menu-item-group__trigger').trigger('click')
  expect(handleGroupChange).toHaveBeenCalledTimes(1)

  wrapper.destroy()
})

test('test menu item group pass render function', async () => {
  const component = {
    components: {
      [Menu.name]: Menu,
      [MenuItemGroup.name]: MenuItemGroup,
      [MenuItem.name]: MenuItem,
    },
    methods: {
      renderIcon(h) {
        return h('div', 'custom icon')
      },

      renderLabel(h) {
        return h('div', 'custom label')
      },
    },
    template: `
      <m-menu>
        <m-menu-item-group name="1" :label="renderLabel" :icon="renderIcon">
          <m-menu-item name="1-1" label="level 1-1" icon="error" />
        </m-menu-item-group>
      </m-menu>
    `,
  }

  const wrapper = mount(component)
  expect(wrapper.html()).toMatchSnapshot()

  wrapper.destroy()
})

test('test menu item group icon pass object', async () => {
  const component = {
    components: {
      [Menu.name]: Menu,
      [MenuItemGroup.name]: MenuItemGroup,
      [MenuItem.name]: MenuItem,
    },
    template: `
      <m-menu>
        <m-menu-item-group name="1" label="level 1" :icon="{ name: 'error', color: 'red', size: 30 }">
          <m-menu-item name="1-1" label="level 1-1" icon="error" />
        </m-menu-item-group>
      </m-menu>
    `,
  }

  const wrapper = mount(component)
  expect(wrapper.html()).toMatchSnapshot()

  wrapper.destroy()
})

test('test menu item group slots', async () => {
  const component = {
    components: {
      [Menu.name]: Menu,
      [MenuItemGroup.name]: MenuItemGroup,
      [MenuItem.name]: MenuItem,
    },
    template: `
      <m-menu>
        <m-menu-item-group name="1">
          <template #icon>
            <div>custom icon</div>
          </template>

          <template #label>
            <div>custom label</div>
          </template>

          <m-menu-item name="1-1" label="level 1-1" icon="error" />
        </m-menu-item-group>
      </m-menu>
    `,
  }

  const wrapper = mount(component)
  expect(wrapper.html()).toMatchSnapshot()

  wrapper.destroy()
})
