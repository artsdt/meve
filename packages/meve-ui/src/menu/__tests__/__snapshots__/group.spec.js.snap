// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test menu item group disabled 1`] = `
"<div class=\\"m-menu m--box\\">
  <div class=\\"m-menu-item-group m--box\\">
    <div class=\\"m-menu-item-group__trigger m-menu-item-group--disabled\\">
      <div class=\\"m-menu-item-group__indent\\" style=\\"padding-left: -24px;\\"></div><i menu-item-group-cover=\\"true\\" class=\\"m-icon m--box meve-icon--set meve-icon-error m-menu-item-group__icon\\"></i>
      <div class=\\"m-menu-item-group__label\\">level 1</div><i menu-item-group-cover=\\"true\\" class=\\"m-icon m--box meve-icon--set meve-icon-menu-down m-menu-item-group__arrow\\"></i>
    </div>
    <div class=\\"m-menu-item-group__list m-menu-item-group--list-height-transition\\" style=\\"height: 0px;\\">
      <div class=\\"m-menu-item m--box\\">
        <div class=\\"m-menu-item__indent\\" style=\\"padding-left: 0px;\\"></div><i menu-item-cover=\\"true\\" class=\\"m-icon m--box meve-icon--set meve-icon-error m-menu-item__icon\\"></i>
        <div class=\\"m-menu-item__label\\">level 1-1</div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`test menu item group icon pass object 1`] = `
"<div class=\\"m-menu m--box\\">
  <div class=\\"m-menu-item-group m--box\\">
    <div class=\\"m-menu-item-group__trigger\\">
      <div class=\\"m-menu-item-group__indent\\" style=\\"padding-left: -24px;\\"></div><i menu-item-group-cover=\\"true\\" class=\\"m-icon m--box meve-icon--set meve-icon-error m-menu-item-group__icon\\" style=\\"color: red; font-size: 30px;\\"></i>
      <div class=\\"m-menu-item-group__label\\">level 1</div><i menu-item-group-cover=\\"true\\" class=\\"m-icon m--box meve-icon--set meve-icon-menu-down m-menu-item-group__arrow\\"></i>
    </div>
    <div class=\\"m-menu-item-group__list m-menu-item-group--list-height-transition\\" style=\\"height: 0px;\\">
      <div class=\\"m-menu-item m--box\\">
        <div class=\\"m-menu-item__indent\\" style=\\"padding-left: 0px;\\"></div><i menu-item-cover=\\"true\\" class=\\"m-icon m--box meve-icon--set meve-icon-error m-menu-item__icon\\"></i>
        <div class=\\"m-menu-item__label\\">level 1-1</div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`test menu item group pass render function 1`] = `
"<div class=\\"m-menu m--box\\">
  <div class=\\"m-menu-item-group m--box\\">
    <div class=\\"m-menu-item-group__trigger\\">
      <div class=\\"m-menu-item-group__indent\\" style=\\"padding-left: -24px;\\"></div>
      <div>custom icon</div>
      <div>custom label</div><i menu-item-group-cover=\\"true\\" class=\\"m-icon m--box meve-icon--set meve-icon-menu-down m-menu-item-group__arrow\\"></i>
    </div>
    <div class=\\"m-menu-item-group__list m-menu-item-group--list-height-transition\\" style=\\"height: 0px;\\">
      <div class=\\"m-menu-item m--box\\">
        <div class=\\"m-menu-item__indent\\" style=\\"padding-left: 0px;\\"></div><i menu-item-cover=\\"true\\" class=\\"m-icon m--box meve-icon--set meve-icon-error m-menu-item__icon\\"></i>
        <div class=\\"m-menu-item__label\\">level 1-1</div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`test menu item group props 1`] = `
"<div class=\\"m-menu m--box\\">
  <div class=\\"m-menu-item-group m--box\\">
    <div class=\\"m-menu-item-group__trigger\\">
      <div class=\\"m-menu-item-group__indent\\" style=\\"padding-left: -24px;\\"></div><i menu-item-group-cover=\\"true\\" class=\\"m-icon m--box meve-icon--set meve-icon-error m-menu-item-group__icon\\"></i>
      <div class=\\"m-menu-item-group__label\\">level 1</div><i menu-item-group-cover=\\"true\\" class=\\"m-icon m--box meve-icon--set meve-icon-menu-down m-menu-item-group__arrow\\"></i>
    </div>
    <div class=\\"m-menu-item-group__list m-menu-item-group--list-height-transition\\" style=\\"height: 0px;\\">
      <div class=\\"m-menu-item m--box\\">
        <div class=\\"m-menu-item__indent\\" style=\\"padding-left: 0px;\\"></div><i menu-item-cover=\\"true\\" class=\\"m-icon m--box meve-icon--set meve-icon-error m-menu-item__icon\\"></i>
        <div class=\\"m-menu-item__label\\">level 1-1</div>
      </div>
    </div>
  </div>
</div>"
`;

exports[`test menu item group slots 1`] = `
"<div class=\\"m-menu m--box\\">
  <div class=\\"m-menu-item-group m--box\\">
    <div class=\\"m-menu-item-group__trigger\\">
      <div class=\\"m-menu-item-group__indent\\" style=\\"padding-left: -24px;\\"></div>
      <div>custom icon</div>
      <div>custom label</div><i menu-item-group-cover=\\"true\\" class=\\"m-icon m--box meve-icon--set meve-icon-menu-down m-menu-item-group__arrow\\"></i>
    </div>
    <div class=\\"m-menu-item-group__list m-menu-item-group--list-height-transition\\" style=\\"height: 0px;\\">
      <div class=\\"m-menu-item m--box\\">
        <div class=\\"m-menu-item__indent\\" style=\\"padding-left: 0px;\\"></div><i menu-item-cover=\\"true\\" class=\\"m-icon m--box meve-icon--set meve-icon-error m-menu-item__icon\\"></i>
        <div class=\\"m-menu-item__label\\">level 1-1</div>
      </div>
    </div>
  </div>
</div>"
`;
